chsdi/lib/helpers.py

```diff
@@ -289,15 +292,29 @@ def parse_box2d(stringBox2D):
     return map(float, extent.split(' '))
 
 
-def center_from_box2d(box2D):
+def is_box2d(box2D):
     # Bottom left to top right only
-    if box2D[0] > box2D[2] or box2D[1] > box2D[3]:
+    if box2D[0] > box2D[2] or box2D[1] > box2D[3] or len(box2D) != 4:
         raise ValueError('Invalid box2D.')
+    return True
+
+
+def extend_box2d(box2D, distance):
+    if is_box2d(box2D):
+        return [
+            box2D[0] - distance,
+            box2D[1] - distance,
+            box2D[2] + distance,
+            box2D[3] + distance
+        ]
 
-    return [
-        box2D[0] + ((box2D[2] - box2D[0]) / 2),
-        box2D[1] + ((box2D[3] - box2D[1]) / 2)
-    ]
+
+def center_from_box2d(box2D):
+    if is_box2d(box2D):
```

Comment:
Well, this has been like this before this PR. But the ValueError raised from is_box2d is not handled in code. see [1] and [2]. Here's an example link of what I mean: http://mf-chsdi3.dev.bgdi.ch/gal_gatilegrid_grid_support//rest/services/inspire/SearchServer?searchText=berges&type=locations&bbox=551754.125,167918.328125,551306.5625,168514.625

It should fail not with internal server error, but with proper 404 error (bad parameter). But this could be done in separate PR.

[1] https://github.com/geoadmin/mf-chsdi3/blob/master/chsdi/views/historicalmaps.py#L31
[2] https://github.com/geoadmin/mf-chsdi3/blob/master/chsdi/views/search.py#L274